core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -47,26 +47,26 @@ import scala.collection.JavaConverters._
  *                                     +----------------------+
  */
 object Scheduler {
-  type SpecInput = (SpecsSnapshot, Source[SpecUpdated, Any])
 
   type StateOutput = (StateSnapshot, Source[StateEvent, Any])
 
-  def fromClient(client: MesosClient): Flow[SpecInput, StateOutput, NotUsed] = {
+  def fromClient(client: MesosClient, specsSnapshot: SpecsSnapshot): Flow[SpecUpdated, StateOutput, NotUsed] = {
```

Comment:
This is way simpler to use and should have the same effect of keeping the ordering.